src/clinical/clinical-api.ts

```diff
@@ -127,6 +134,40 @@ class ClinicalController {
     res.status(200).json(entityData);
   }
 
+  @HasProgramReadAccess((req: Request) => req.params.programId)
+  async getProgramClinicalSearchResults(req: Request, res: Response) {
+    const programId: string = req.params.programId;
+    const state: CompletionStates = req.query.completionState || CompletionStates.all;
+    const completionState: {} = completionFilters[state] || {};
+    const entityTypes: string[] =
+      req.query.entityTypes && req.query.entityTypes.length > 0
+        ? req.query.entityTypes.split(',')
+        : [''];
+
+    // FE filters digits out of search text for Donor search
+    const donorIds = req.query.donorIds.match(/\d*/gi)?.filter((match: string) => !!match) || [];
+    const submitterDonorIds =
+      req.query.submitterDonorIds && req.query.submitterDonorIds.length > 0
+        ? req.query.submitterDonorIds.split(',').filter((match: string) => !!match)
```

Comment:
little trick i like if you're filtering out just passed on truthy
```...split(',').filter(Boolean)```
this will coerce `match` into a boolean and filter accordingly

Reply:
Awesome I've updated here and will use going forward, the more concise the better!